### ISSUE TITLE
Disabled transparent buttons should stay transparent and "disable" text inside.

### ISSUE THREAD

**Issue by kodayashi** [GeekyAnts/NativeBase#1742]
Hi! Thanks for trying out NativeBase!
Due to the overwhelming requests and issues/questions we ask you to add the following details with your issue.

Please check the existing open/closed issues for a possible duplicate before creating a new issue :)

Also **DO** fill out the below form to give us a better idea about your environment and help us debug it quicker. Issues without the required details will mostly be unattended to.

## react-native, react and native-base version
"react-native": "0.53.3",
"native-base": "^2.3.9",

## Expected behaviour
The button should stay transparent and disable the text inside it.

## Actual behaviour
The button's non-transparent disable style (gray) is applied and the transparent button is no longer transparent.

## Steps to reproduce (code snippet or screenshot)
Create a transparent button with text, set disabled to true

## Screenshot of emulator/device
N/A

## Is the bug present in both ios and android or in any one of them?
At least iOS

## Any other additional info which would help us debug the issue quicker.
N/A

**Comment by kodayashi** [comment]
FWIW, I changed the buttonTheme's "disabled" style to work around this. I change it from

`  const buttonTheme = {
    '.disabled': {
      backgroundColor: variables.btnDisabledBg,
    },`

to

`  const buttonTheme = {
    '.disabled': {
      '.transparent': {
        backgroundColor: 'transparent',
        'NativeBase.Text': {
          color: variables.btnDisabledBg,
        },
      },
      backgroundColor: variables.btnDisabledBg,
    },`

**Comment by akhil-geekyants** [comment]
@kodayashi try placing transparent prop at end
like
```
 <Button disabled transparent>
      <Text>Primary</Text>
 </Button>
```

**Comment by SupriyaKalghatgi** [comment]
Fixed with [2.4.2](https://github.com/GeekyAnts/NativeBase/releases/tag/v2.4.2)
